diff --git a/lib/ExprToBDDTransformer.cpp b/lib/ExprToBDDTransformer.cpp
index e4503be..e807443 100644
--- a/lib/ExprToBDDTransformer.cpp
+++ b/lib/ExprToBDDTransformer.cpp
@@ -482,7 +482,7 @@ BDDInterval ExprToBDDTransformer::getBDDFromExpr(const expr &e, const vector<bou
 	}
 
 	BDDInterval bodyBdd;
-	if (onlyExistentials)
+	if (false && onlyExistentials)
 	{
 	    if (Z3_is_quantifier_forall(*context, ast))
 	    {
@@ -1147,7 +1147,7 @@ Model ExprToBDDTransformer::GetModel(BDD modelBdd)
 
 	const auto varSort = varSorts.find(name)->second;
 	if (varSort.is_bool()) {
-	    model.insert({name, modelBV[0]});
+	    model.insert({name, (bool)modelBV[0]});
 	} else {
 	    model.insert({name, modelBV});
 	}
diff --git a/lib/ExprToBDDTransformer.h b/lib/ExprToBDDTransformer.h
index c255240..9e8d25a 100644
--- a/lib/ExprToBDDTransformer.h
+++ b/lib/ExprToBDDTransformer.h
@@ -29,7 +29,7 @@ using namespace cudd;
 
 class ExprToBDDTransformer
 {
-  private:
+  public:
     Cudd bddManager;
 
     std::map<std::string, Bvec> vars;
